src/main/java/ch/njol/skript/util/Date.java

```diff
@@ -1,142 +1,140 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.util;
 
+import ch.njol.skript.SkriptConfig;
+import ch.njol.yggdrasil.YggdrasilSerializable;
+import org.jetbrains.annotations.Nullable;
+
 import java.util.TimeZone;
 
-import org.jetbrains.annotations.Nullable;
+public class Date extends java.util.Date implements YggdrasilSerializable {
 
-import ch.njol.skript.SkriptConfig;
-import ch.njol.yggdrasil.YggdrasilSerializable;
+	/**
+	 * Get a new Date with the current time
+	 *
+	 * @return New date with the current time
+	 */
+	public static Date now() {
+		return new Date();
+	}
+
+	/**
+	 * Converts a {@link java.util.Date} to a {@link Date}.
+	 *
+	 * @param date The {@link java.util.Date} to convert.
+	 * @return The converted date.
+	 */
+	public static Date fromJavaDate(java.util.Date date) {
+		return new Date(date.getTime());
+	}
```

Comment:
```suggestion
	public static Date fromJavaDate(java.util.Date date) {
		if (date instanceof Date ours)
			return ours;
		return new Date(date.getTime());
	}
```
Since we are now a Java date, it's conceivable an argument to this could already be one of our dates, in which case we don't need to create a new copy.